src/main/java/org/craftercms/studio/impl/v2/job/StudioPublisherTask.java

```diff
@@ -0,0 +1,431 @@
+                                List<PublishRequest> itemsToDeploy =
+                                        publishingManager.getItemsReadyForDeployment(siteId, environment);
+                                while (CollectionUtils.isNotEmpty(itemsToDeploy)) {
+                                    logger.debug("Deploying " + itemsToDeploy.size() + " items for " +
+                                            "site " + siteId);
+                                    publishingManager.markItemsProcessing(siteId, environment, itemsToDeploy);
+                                    List<String> commitIds = itemsToDeploy.stream()
+                                            .map(PublishRequest::getCommitId)
+                                            .distinct().collect(Collectors.toList());
+
+                                    boolean allCommitsPresent = true;
+                                    for (String commit : commitIds) {
+                                        if (StringUtils.isNotEmpty(commit)) {
+                                            boolean commitPresent = contentRepository.commitIdExists(siteId,
+                                                    commit);
+                                            if (!commitPresent) {
+                                                logger.debug("Commit with ID: " + commit + " is not present in " +
+                                                        "local repo for site " + siteId + ". " +
+                                                        "Publisher task will skip this cycle.");
+                                                allCommitsPresent = false;
+                                            }
+                                        }
+                                    }
+
+                                    if (allCommitsPresent) {
+                                        logger.info("Starting publishing on environment " + environment +
+                                                " for site " + siteId);
+                                        logger.debug("Site \"{0}\" has {1} items ready for deployment",
+                                                siteId, itemsToDeploy.size());
+
+                                        doPublishing(siteId, itemsToDeploy, environment);
+                                        retryCounter.remove(siteId);
+                                    } else {
+                                        publishingManager.markItemsReady(siteId, environment, itemsToDeploy);
+                                        int retriesLeft = retryCounter.get(siteId) - 1;
+                                        if (retriesLeft > 0) {
+                                            retryCounter.put(siteId, retriesLeft);
+                                        } else {
+                                            siteService.enablePublishing(siteId, false);
```

Comment:
Add a statement to remove the retry counter from the map.

Reply:
Throw an exception with the correct error message indicating the problem of file not present on disk.